lib/StringClient.js

```diff
@@ -92,10 +92,12 @@ StringClient.prototype.read = function read(options, callback) {
     var self = this;
     this.request(options, function _parse(err, req) {
         if (err) {
-            return (callback(err, req));
+            // in the case of an error here, make sure to still call onResult.
```

Comment:
It was, the comment here is probably misleading. 

I just meant to call out that _onResult is now the unified "exit" point for StringClient and other classes that inherit from it, so that we should still call _onResult (vs calling callback directly and bypassing the exit point).

Reply:
I think what I meant is that in case of an error being passed to this callback, if `'result'` is still emitted on the `req` object, then maybe we don't need to call `_onResult` manually and just let the `'result'` handler below handle this case too to avoid this special case?

Not sure if that'd be better, just thinking out loud.